### DISCUSSION TITLE
x-model not masked in x-mask

### DISCUSSION THREAD

**Discussion by baravak** [alpinejs/alpine#2863] [7. Bugs (archived)]
if use x-model, input not masked
``` html
<input x-data="{amount: 132558}" x-mask:dynamic="$money($input)" x-model="amount">
```
![image](https://user-images.githubusercontent.com/953471/164714546-941fc4d8-691c-4c0c-b985-2ae538bc2722.png)

**Comment by benkoshy** [comment]
A reproducible example would be inordinately helpful.

**Reply by JanderV** [reply]
It does work but not for the initial value: https://codepen.io/restartz1/pen/KKBOvMv

I think changing the `processInputValue(el)` calls in the mask plugin code to the following should fix it:
```
queueMicrotask(() => {
       processInputValue(el)
 })
```

Update:
Here's [the pull request](https://github.com/alpinejs/alpine/pull/3414)
